### PR TITLE
Add PyPi pipeline for "kvakk-git-tools" and gitconfig validation

### PR DESCRIPTION
**Pull request by Andilun**
- Set up a PyPi deployment pipeline for the new "kvakk-git-tools" package.
- Remove dash from ssb-gitconfig.
- Add validation for SSB recommended git configuration based on platform.
- Add tests for git configuration validation.

BREAKING CHANGE:
Moving the recommended folder inside the package has the side effect of breaking old versions of the script. Old versions will no longer find the recommended configuration files in main.

### CODE REVIEW

**Review by bjornandre — APPROVED** [review]
Looks good to me 👍

File: kvakk_git_tools/ssb_gitconfig.py [hunk 1]
```diff
@@ -208,22 +211,22 @@ def set_base_config(pl: Platform, test: bool) -> str:
...
     temp_dir = Path.home() / "temp-ssb-gitconfig"
-    config_dir = temp_dir / "kvakk-git-tools" / "recommended"
+    config_dir = temp_dir / "kvakk-git-tools" / "kvakk_git_tools" / "recommended"
     dst = Path().home() / ".gitconfig"
     src = config_dir / f"gitconfig-{pl.name().value}"
     if test:
         src = config_dir / "gitconfig-dapla"
 
-    options = []
+    options = ["--branch", "2.0.0"]
```
Comment (arneso-ssb) [inline kvakk_git_tools/ssb_gitconfig.py:220]: @Andilun The 2.0.0-branch will be removed when merging to main, right? Then you need to clear this option.

**Review by arneso-ssb — CHANGES_REQUESTED** [review]
Looks good, but need to fix the --branch 2.0.0 option if this branch is not a permanent branch.

File: kvakk_git_tools/ssb_gitconfig.py (hunk 1 above)
Comment (Andilun) [reply kvakk_git_tools/ssb_gitconfig.py:220]: 2.0.0 is a tag.

`git clone https://github.com/statisticsnorway/kvakk-git-tools.git --branch 2.0.0`

Clones the 2.0.0 tag of the repo. This version locks the script such that future changes to project structure wont brake the script again.

**Pull request merged**